stdlib/Distributed/src/managers.jl

```diff
@@ -461,47 +461,42 @@ end
 
 const client_port = Ref{Cushort}(0)
 
-function socket_reuse_port()
+function socket_reuse_port(iptype)
     if ccall(:jl_has_so_reuseport, Int32, ()) == 1
-        s = TCPSocket(delay = false)
+        sock = TCPSocket(delay = false)
 
         # Some systems (e.g. Linux) require the port to be bound before setting REUSEPORT
         bind_early = Sys.islinux()
 
-        bind_early && bind_client_port(s)
-        rc = ccall(:jl_tcp_reuseport, Int32, (Ptr{Cvoid},), s.handle)
+        bind_early && bind_client_port(sock, iptype)
+        rc = ccall(:jl_tcp_reuseport, Int32, (Ptr{Cvoid},), sock.handle)
         if rc < 0
             # This is an issue only on systems with lots of client connections, hence delay the warning
             nworkers() > 128 && @warn "Error trying to reuse client port number, falling back to regular socket" maxlog=1
 
             # provide a clean new socket
             return TCPSocket()
         end
-        bind_early || bind_client_port(s)
-        return s
+        bind_early || bind_client_port(sock, iptype)
+        return sock
     else
         return TCPSocket()
     end
 end
 
-# TODO: this doesn't belong here, it belongs in Sockets
-function bind_client_port(s::TCPSocket)
-    Sockets.iolock_begin()
-    @assert s.status == Sockets.StatusInit
-    host_in = Ref(hton(UInt32(0))) # IPv4 0.0.0.0
-    err = ccall(:jl_tcp_bind, Int32, (Ptr{Cvoid}, UInt16, Ptr{Cvoid}, Cuint, Cint),
-                s, hton(client_port[]), host_in, 0, false)
-    Sockets.iolock_end()
-    uv_error("tcp_bind", err)
-
-    _addr, port = getsockname(s)
+function bind_client_port(sock::TCPSocket, iptype)
+    bind_host = iptype == IPv4 ? IPv4(0) : IPv6(0)
```

Comment:
Use `===` to compare? Also, why not use use the type constructor directly, e.g. `iptype(0)`?

Reply:
changed to `iptyoe(0)` in ad140ac